code/io/increaseHumanGEMVersion.m

```diff
@@ -1,79 +1,107 @@
-%   bumpType      the value has to be one selected among 'major', 'minor' or 'patch'
+%   bumpType    either 'major', 'minor' or 'patch'
+%   test        true if a test run should be done that can take in a
+%               development branch (it will not check if current branch is
+%               main, nor update the model version, history and readme files).
 %
 % Usage: increaseHumanGEMVersion(bumpType)
 %
-
-
-%Check if in main:
-currentBranch = git('rev-parse --abbrev-ref HEAD');
-if ~strcmp(currentBranch,'main')
-    error('ERROR: not in main')
+if nargin<2
+    test=false;
 end
 
 %Get model path
 [ST, I]=dbstack('-completenames');
 modelPath=fileparts(fileparts(fileparts(ST(I).file)));
 
-%Bump version number:
-versionFile=fullfile(modelPath,'version.txt');
-fid = fopen(versionFile,'r');
+%Check if in main:
+if ~test
+    currentBranch = git('rev-parse --abbrev-ref HEAD');
+    if ~strcmp(currentBranch,'main')
+        error('ERROR: not in main')
+    end
+
+    %Bump version number:
+    versionFile=fullfile(modelPath,'version.txt');
+    fid = fopen(versionFile,'r');
     oldVersion = fscanf(fid, '%s');
-fclose(fid);
-oldVersion = str2double(strsplit(oldVersion,'.'));
-newVersion = oldVersion;
-switch bumpType
-    case 'major'
-        newVersion(1) = newVersion(1) + 1;
-        newVersion(2) = 0;
-        newVersion(3) = 0;
-    case 'minor'
-        newVersion(2) = newVersion(2) + 1;
-        newVersion(3) = 0;
-    case 'patch'
-        newVersion(3) = newVersion(3) + 1;
-    otherwise
-        error('ERROR: invalid input. Use either "major", "minor" or "patch"')
-end
-newVersion = num2str(newVersion,'%d.%d.%d');
+    fclose(fid);
+    oldVersion = str2double(strsplit(oldVersion,'.'));
+    newVersion = oldVersion;
+    switch bumpType
+        case 'major'
+            newVersion(1) = newVersion(1) + 1;
+            newVersion(2) = 0;
+            newVersion(3) = 0;
+        case 'minor'
+            newVersion(2) = newVersion(2) + 1;
+            newVersion(3) = 0;
+        case 'patch'
+            newVersion(3) = newVersion(3) + 1;
+        otherwise
+            error('ERROR: invalid input. Use either "major", "minor" or "patch"')
+    end
+    newVersion = num2str(newVersion,'%d.%d.%d');
 
-%Check if history has been updated:
-%fid     = fopen('../../history.md','r');
-%history = fscanf(fid,'%s');
-%fclose(fid);
-%if ~contains(history,['human' newVersion ':'])
-%    error('ERROR: update history.md first')
-%end
-%To be included
+    %Check if history has been updated:
+    fid     = fopen(fullfile(modelPath,'history.md'),'r');
+    history = fscanf(fid,'%s');
+    fclose(fid);
+    if ~contains(history,['human' newVersion ':'])
+        error('ERROR: update history.md first')
+    end
+end
 
 %Load model:
-ymlFile=fullfile(modelPath,'model','Human-GEM.yml');
-ihuman = importYaml(ymlFile);
+currDir = pwd;
+cd(fullfile(modelPath,'code','io'))
+ihuman = readYAMLmodel_HumanGEM(fullfile(modelPath,'model','Human-GEM.yml'));
```

Comment:
I noticed that in the README.md, it mentions using readYAMLmodel to load the model, but the function readYAMLmodel_HumanGEM is also mentioned. Could you clarify whether both methods should be used for loading the model, or if there is a specific context for using one over the other?  The same also for export.

Reply:
It's a temporary solution, until https://github.com/SysBioChalmers/RAVEN/pull/575 has made it into a new release (= soon). After that, the `_HumanGEM` version will be removed (it is identical to the updated RAVEN function). We can also wait a few days to have that resolved, before merging this PR.

Is everything else working fine?

Reply:
Besides this, everything works fine here. But the export is super slow. Do you think there is any need to ask @JHL-452b  to test this too?

Reply:
It takes about the same time as the function did in release 1.19.0, as about 80% of the time is due to writing the SBML file (combination of having a large model and libSBML being inefficient.

RAVEN 2.10.3 is now released, so the `_HumanGEM` I/O functions are removed from this PR. Also keep in mind that this function is only used when a new release is made. If it worked for you, then I do not see reason for further reviewing.